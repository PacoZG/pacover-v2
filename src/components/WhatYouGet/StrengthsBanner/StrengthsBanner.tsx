import React from 'react'
import { useTranslations } from 'next-intl'

type StrengthsBannerProps = {
  StrengthsIcon: React.ElementType
  translationKey: string
}

const StrengthsBanner: React.FC<StrengthsBannerProps> = ({ StrengthsIcon, translationKey }) => {
  const t = useTranslations('skills')

  return (
    <div className="strengths-banner">
      <StrengthsIcon />
      <p className="strengths-text">{t(translationKey)}</p>
    </div>
  )
}

export default StrengthsBanner
