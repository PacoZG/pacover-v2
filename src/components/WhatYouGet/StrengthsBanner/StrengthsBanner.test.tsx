import React from 'react'
import { render } from '@testing-library/react'
import { beforeEach, describe, expect, test, vi } from 'vitest'
import StrengthsBanner from '@/components/WhatYouGet/StrengthsBanner/StrengthsBanner'

describe('StrengthsBanner', () => {
  beforeEach(() => {
    window.scrollTo = vi.fn()

    vi.mock('next-intl', () => ({
      useTranslations: () => (key: string) => key,
    }))
  })

  describe('StrengthsBanner component', () => {
    test('matches the snapshot', () => {
      const { container } = render(
        <StrengthsBanner StrengthsIcon={'symbol'} translationKey={''} />
      )

      expect(container).toMatchSnapshot()
    })
  })
})
