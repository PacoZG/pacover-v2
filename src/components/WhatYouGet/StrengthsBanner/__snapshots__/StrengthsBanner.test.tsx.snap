// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StrengthsBanner > StrengthsBanner component > matches the snapshot 1`] = `
<div>
  <div
    class="strengths-banner"
  >
    <symbol />
    <p
      class="strengths-text"
    />
  </div>
</div>
`;
